Add MouseEvent/MouseEventInit as untested DOM IDLs

csnardi opened the pull request:
The DOM idlharness test would not run as `MouseEvent`/`MouseEventInit` were not declared within any IDL files parsed. Add these in as untested IDLs.

annevk commented:
I don't really understand this.

Presumably this is because `html.idl` is imported, but `html/dom/interfaces.https.html` does not include a mention of `MouseEvent` either. So what's going on?

csnardi commented:
`html/dom/interfaces.https.html` pulls in UIEvents as untested IDL, which defines `MouseEvent`/`MouseEventInit`. I could do that as well, but I figured it might just be easier to do it this way.

annevk commented:
If the dependencies are equivalent it seems better to copy HTML's style as that seems more forward compatible.

annevk commented:
Although I guess if someone updates HTML and forgets about DOM we're here again. Hmm.

annevk commented:
cc @kereliuk @foolip

gsnedders commented:
Related: #10347

foolip commented:
@csnardi can you try the mechanism that @lukebjerring introduced in https://github.com/web-platform-tests/wpt/pull/10381? By using `add_dependency_idls(html)` I think the issue might be avoided.

csnardi commented:
Using `add_dependency_idls` with UIEvents worked, so I updated the PR.

foolip commented on dom/interfaces.html:
```diff
@@ -18,11 +18,12 @@ <h1>DOM IDL tests</h1>
   // HTML is needed for EventHandler. Provide a dummy interface for
   // LinkStyle which HTML depends on but we're not testing.
   idlArray.add_untested_idls('interface LinkStyle {};');
   idlArray.add_untested_idls(html);
```
I was wondering if using `add_dependency_idls` *here* would do the trick, and avoid 'interface LinkStyle {};' and the extra bits from uievents entirely.

csnardi replied:
Oh, that makes more sense. That does indeed work.

csnardi committed:
Move HTML from untested IDL to a dependency IDL for DOM

The DOM idlharness test would not run as dependencies for the HTML IDL file were not declared. However, we don't need to define HTML as untested IDL -- using the new dependency IDL feature generates the same tests but prevents the current dependency problem.

foolip reviewed (approved):
LGTM assuming the number of passing/failing tests is unchanged.

csnardi commented:
The number of passing/failing tests was unchanged from my first commit to this -- currently no tests are run due to this dependency issue.

foolip commented:
Awesome, merge away when Travis is happy!
